Flow Framework documentation for no-op step delay (#7080)
opensearch-project/documentation-website: _automating-configurations/workflow-steps.md

|Step type	|Corresponding API	|Description	|
|---	|---	|---	|

> | `noop` |  No API | A no-operation (no-op) step that does nothing, which is useful for synchronizing parallel steps. If the `user_inputs` field contains a `delay` key, this step will wait for the specified time.	|

Reviewer: ```suggestion
| `noop` | No API | A no-operation (no-op) step that does nothing, which is useful for synchronizing parallel steps. If the `user_inputs` field contains a `delay` key, this step will wait for the specified amount of time.	|
```

> | `noop` |  No API | A no-operation (no-op) step that does nothing. A no-op is sometimes useful for synchronizing parallel steps. If the `user_inputs` field contains a `delay` key, this step will wait for the specified time.	|

Reviewer 2: ```suggestion
| `noop` |  No API | A no-operation (no-op) step that does nothing, which is useful for synchronizing parallel steps. If the `user_inputs` field contains a `delay` key, this step will wait for the specified time.	|
```

|`create_connector`	|[Create Connector]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/connector-apis/create-connector/)	|Creates a connector to a model hosted on a third-party platform.	|
|`delete_connector`	|[Delete Connector]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/connector-apis/delete-connector/)	|Deletes a connector to a model hosted on a third-party platform.	|
|`register_model_group`	|[Register Model Group]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-group-apis/register-model-group/)	|Registers a model group. The model group will be deleted automatically once no model is present in the group.	|

> |`register_remote_model`	|[Register Model (remote)]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/#register-a-model-hosted-on-a-third-party-platform)	| Registers a model hosted on a third-party platform. If the `user_inputs` field contains a `deploy` key that is set to `true`, also deploys the model.	| 

Reviewer: ```suggestion
|`register_remote_model`	|[Register Model (remote)]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/#register-a-model-hosted-on-a-third-party-platform)	| Registers a model hosted on a third-party platform. If the `user_inputs` field contains a `deploy` key that is set to `true`, the model is also deployed.	| 
```

|Step type	|Corresponding API	|Description	|
|---	|---	|---	|

> |`noop`	|No API	| A no-operation (no-op) step that does nothing. It may be useful in some cases for synchronizing parallel steps. If the `user_inputs` field contains a `delay` key, this step will wait for the specified time.	|

Reviewer 2: ```suggestion
| `noop` |  No API | A no-operation (no-op) step that does nothing. A no-op is sometimes useful for synchronizing parallel steps. If the `user_inputs` field contains a `delay` key, this step will wait for the specified time.	|
```

## Additional fields

> You can include the following additional fields in the `user_inputs` field when indicated if the field is supported by the indicated step type.

Reviewer: Can "when indicated" be removed here?

> You can include the following additional fields in the `user_inputs` field when indicated if the field is supported by the indicated step type.

Reviewer 2: ```suggestion
You can include the following additional fields in the `user_inputs` field if the field is supported by the indicated step type.
```

|Field	|Data type	| Step type | Description	|
|---	|---	|---	|

> |`node_timeout`	| Time units	| All | A user-provided timeout for this step. For example, `20s` for a 20-second timeout.	|

Author: I like the extra column.  One error, `tools_order` is for "Register Agent" step.

> |`node_timeout`	| Time units	| All | A user-provided timeout for this step. For example, `20s` for a 20-second timeout.	|

Author: Pushed a commit to fix the step name.  This PR LGTM!

> |`deploy`	| Boolean	| Register Model | If set to `true`, also executes the Deploy Model step.	|

Reviewer: ```suggestion
|`deploy`	| Boolean	| Register Model | If set to `true`, also deploys the model.	|
```

> |`deploy`	| Boolean	| Register Model | If set to `true`, also executes the Deploy Model step.	|

Reviewer: Should "Register model" be in sentence case?

> |`tools_order`	| List	| Register Agent | Specifies the ordering of `tools`. For example, specify `["foo_tool", "bar_tool"]` to sequence those tools in that order.	|

Reviewer: Should "Register agent" be in sentence case?

> |`delay`	| Time units	| No-op | Waits the specified time. For example, `250ms` will wait for 250 milliseconds before continuing the workflow.	|

Reviewer: ```suggestion
|`delay`	| Time units	| No-op | Waits for the specified amount of time. For example, `250ms` specifies to wait for 250 milliseconds before continuing the workflow.	|
```

|`node_timeout`	|Time units	|A user-provided timeout for this step. For example, `20s` for a 20-second timeout.	|
|`deploy`	|Boolean	|Applicable to the Register Model step type. If set to `true`, also executes the Deploy Model step.	|
|`tools_order`	|List	|Applicable only to the Register Agent step type. Specifies the ordering of `tools`. For example, specify `["foo_tool", "bar_tool"]` to sequence those tools in that order.	|

> |`delay`	|Time units	|Applicable to the No-op step type. Waits the specified time. For example, `250ms` will wait for 250 milliseconds before continuing the workflow.	|

Reviewer 2: Does this change the technical meaning?

```suggestion
| `delay` | Time units | Sets the specified time to wait during a no-op step.  For example, when set to `250ms`, the no-op step waits for 250 milliseconds before continuing the workflow. |
```

Author: > Does this change the technical meaning?

The steps in this table are only applicable to some steps.  The two entries above this are specific only for the Register Model Step, for example, so we should address these consistently.  Perhaps a new column in the table for "Steps applicable to" ?

Reviewer 2: I think an extra column in the table might make sense. Let me take a pass at it.
